### PR TITLE
Revamped Get Started page

### LINKED ISSUE #15749: Apply design suggestion to Header on Get Started Page
### Is your enhancement related to a problem? Please describe.

**Header**
- [x] Move `subtitle`, `instructions text` and `items count` up so that it is under the title (see design mockup below) on a white background
Screenshot_20200128_163310

### LINKED ISSUE #15776: Footer is not displayed on Get Started page
### Describe the bug
Footer is not displayed on Get Started page.
It's not relevant to every browser.
It's not displayed on Google Chrome (79.0.3945.88 Linux 64 Bit)
Screenshot_20200121_170059
But it's displayed well on Google Chrome (78 Linux 64 Bit) and Firefox
Screenshot_20200121_175948

### LINKED ISSUE #15858: Apply design suggestions to template cards of Get Started Page
### Is your task related to a problem? Please describe.
We introduced Get Started page but it should be improved:

- [x] Remove white from behind cards so they are on a grey background (see design mockup below for reference)
- [x] Remove "Create & Open" buttons and create and start workspace as soon as user clicks on Template. A new workspace is supposed to be created and started as soon as the user clicks on the Template card. 
When hovering over a card, the card’s shadow turns from small to large. On click, create and open the workspace. Note: The visual treatment on click should not change (remove the blue border on click).
Example of hover shadow:
image

- [x] Reduce Font size on card items 
    - Sample name - 16px
    - Copy - 14px
Screenshot_20200128_163607

### LINKED ISSUE #15859: Apply design suggestion to Search on Get Started Page
### Is your task related to a problem? Please describe.
We introduced Get Started page but it should be improved:
- [x] Change the text from “Search” to “Filter by...” on that form input control
- [x] Remove the magnifying glass icon and align text left
- [x] Move search/filter control to left side of screen 
- [x] Change the color of the input field border from blue to black. The color of the border on existing forms on other screens form fields is better. Blue rule calls too much attention to the control and this should not stand out so much.
Screenshot_20200128_163409
Border on this form is better^^

See 
Screenshot_20200128_163758

### LINKED ISSUE #15978: [UD] Create button component styled as PatternFly 4 Secondary button
### Is your task related to a problem? Please describe.

The design of Get Started page was developed using PatternFly 4 components, styles and typography.
Currently "Create a Custom Workspace" button doesn't look and feel as PatternFly 4 Secondary button. 

*Current button*
Screenshot 2020-02-10 at 18 56 41

*Original design button*
Screenshot 2020-02-10 at 18 47 18

### Describe the solution you'd like

Create new button component based on PatternFly 4 Secondary button.

### Describe alternatives you've considered

Apply PF4 styles to existing `che-button` widget. 
Applying correct styles on top of existing widgets is tiresome and error-prone process. This will result in mess of PatternFly, MaterialDesign and other styles.

### LINKED ISSUE #15980: [UD] Create masthead component
### Is your task related to a problem? Please describe.

The design of Get Started page was developed using PatternFly 4 components, styles and typography.
But existing `che-toolbar` component on this page differs from masthead provided in original design doc. Correct spacing and typography should be used in page toolbar.

### Describe the solution you'd like

Create new masthead component that follows PatternFly 4 guidelines. 

### Describe alternatives you've considered

Apply PF4 styles to existing `che-toolbar` component.
Applying correct styles on top of existing components is tiresome and error-prone process. This will result in mess of PatternFly, MaterialDesign and other styles.

### LINKED ISSUE #15981: [UD] Create a toolbar component
### Is your task related to a problem? Please describe.

In accordance to the design of Get Started page we need to have a toolbar component that re-uses PatternFly 4 toolbar styles and capabilities.

Screenshot 2020-02-10 at 19 33 35

### Describe the solution you'd like

Create new toolbar component based on PatternFly 4 toolbar component.

### Describe alternatives you've considered

N/A

### LINKED ISSUE #15985: [UD] Create content section component
### Is your task related to a problem? Please describe.

Since the design of Get Started page was developed using PatternFly 4 components, styles and typography, we need to provide a new component to handle a page content in accordance to PF4 guidelines.

### Describe the solution you'd like

Create new content section component.

### Describe alternatives you've considered

Apply PF4 styles to the material design `md-content` component.
Applying correct styles on top of existing widgets is tiresome and error-prone process. This will result in mess of PatternFly, MaterialDesign and other styles.

### LINKED ISSUE #16241: Finalize get-started page redesigning
### Is your task related to a problem? Please describe.
The main work to redesign Get Started page is done but the PR https://github.com/eclipse/che/pull/16157
but it's not merged yet for two reasons:
1. PR is not reviewed and approved yet
2. PR depends on CQ https://dev.eclipse.org/ipzilla/show_bug.cgi?id=21722

So, this issue is about fixing possible reviewers comments and merge PR once CQ is approved

### PR DESCRIPTION
**Pull request by akurinnoy**
### What does this PR do?

This PR introduces revamped Get Started page which takes into account all suggestions referenced in section below.

Screenshot 2020-02-27 at 12 18 46

<details><summary>Beware! Animated gif inside!</summary>

Screen Recording 2020-02-27 at 12 39 27 2020-02-27 12_49_54

</details>

### What issues does this PR fix or reference?

fixes https://github.com/eclipse/che/issues/15858
fixes https://github.com/eclipse/che/issues/15859
fixes https://github.com/eclipse/che/issues/15749
fixes https://github.com/eclipse/che/issues/15776
resolves https://github.com/eclipse/che/issues/15978
resolves https://github.com/eclipse/che/issues/15981
resolves https://github.com/eclipse/che/issues/15985
resolves https://github.com/eclipse/che/issues/15980

resolves https://github.com/eclipse/che/issues/16241

CQ for new dependency https://dev.eclipse.org/ipzilla/show_bug.cgi?id=21783

### CODE REVIEW

**Comment by akurinnoy** [comment]
@beaumorley could you take a look at screenshot and animated gif and say whether all your suggestions were taken into account?

**Review by olexii4 — APPROVED** [review]

File: dashboard/src/components/che-pf-widget/text-input/che-pf-text-input.spec.ts [hunk 1]
```diff
@@ -0,0 +1,300 @@
...
+
+      expect(compiledDirective.find('input').val()).toEqual(value);
+    });
+
+  });
+
+  describe('ngChange > ', () => {
+
+    // todo fix this test
+    xit('should be called on ngModel changes > ', () => {
```
Comment (olexii4) [inline dashboard/src/components/che-pf-widget/text-input/che-pf-text-input.spec.ts:216]: todo ->  TODO

Comment (olexii4) [reply dashboard/src/components/che-pf-widget/text-input/che-pf-text-input.spec.ts:216]: Or just remove this test

**Comment by sleshchenko** [comment]
I tested happy path dockerimage and seems get started page is broken Screenshot_20200227_180342

**Comment by olexii4** [comment]
@sleshchenko should be OK after this commit  **e446540**

**Comment by olexii4** [comment]
[ci-test]

**Comment by akurinnoy** [comment]
rebased to origin master

**Comment by olexii4** [comment]
[ci-test]

**Comment by olexii4** [comment]
[ci-test]

**Comment by sleshchenko** [comment]
crw-ci-test

**Comment by sleshchenko** [comment]
Testing on https://che-sleshche.apps.ocp44.codereadyqe.com/dashboard/#/getstarted
Footer seems to be missing on Chrome and Firefox as well.
Screenshot_20200318_150202

**Review by sleshchenko — APPROVED** [review]
Can't wait to see this PR merged.

**Pull request merged**